test/test_repository/test_project_repository.py

```diff
@@ -182,16 +183,29 @@ def test_save(self):
 
 
     @with_context
-    def test_save(self):
-        """Test save does not fail if the project has no password"""
+    def test_save_project_password_required(self):
+        """Test save project without password fails when password required"""
+        with patch.dict(self.flask_app.config,
+                        {'PROJECT_PASSWORD_REQUIRED': True}):
+            project = ProjectFactory.build()
+            assert self.project_repo.get(project.id) is None
 
-        project = ProjectFactory.build()
-        assert self.project_repo.get(project.id) is None
+            self.project_repo.save(project)
 
-        self.project_repo.save(project)
+            assert_raises(BadRequest, self.project_repo.save, project)
 
-        assert self.project_repo.get(project.id) == project, "Project not saved"
 
+    @with_context
+    def test_save_project_password_not_required(self):
+        """Test save project without password success when password not required"""
+        with patch.dict(self.flask_app.config,
+                        {'PROJECT_PASSWORD_REQUIRED': False}):
+            project = ProjectFactory.build()
+            assert self.project_repo.get(project.id) is None
+
+            self.project_repo.save(project)
+
+            assert self.project_repo.get(project.id) == project, "Project not saved"
```

Comment:
If the password is not required and not provided, shouldn't the project save succeed?